types/inlineCompletionWithReferences.ts

```diff
@@ -58,6 +66,12 @@ export interface InlineCompletionStates {
      * Recommendation was filtered out on the client-side and marked as discarded.
      */
     discarded: boolean
+
+    /**
+     * POC-NEP: Flag to indicate if this was an edit suggestion rather than an inline completion.
+     * This is used by the Next Edit Prediction feature for telemetry and to improve future suggestions.
+     */
+    isEdit?: boolean
```

Comment:
This isn't a completion state and doesn't belong here.  It looks like InlineCompletionStates is used in LogInlineCompletionSessionResultsParams below.  I think we shouldn't pass the flag for edit back and forth between the client and server here at all.  When the server generates a sessionId, it should keep up with the state of whether the associated suggestion was an edit or not.  That way it's entirely a server implementation detail and reduces complexity.